packages/plugins/java/java/src/visitor.ts

```diff
@@ -262,13 +262,14 @@ ${getters}
   }
 
   InputObjectTypeDefinition(node: InputObjectTypeDefinitionNode): string {
-    const name = `${this.convertName(node)}Input`;
+    const convertedName = this.convertName(node);
+    const name = convertedName.endsWith('Input') ? convertedName : `${convertedName}Input`;
```

Comment:
This is the crux of the change - omit the "Input" suffix if the schema name already includes it